libdnf/module/modulemd/ModuleDependencies.cpp

```diff
@@ -0,0 +1,52 @@
+#include <modulemd/modulemd-simpleset.h>
+#include <sstream>
+
+#include "libdnf/utils/utils.hpp"
+#include "ModuleDependencies.hpp"
+
+ModuleDependencies::ModuleDependencies(ModulemdDependencies *dependencies)
+        : dependencies(dependencies)
+{}
+
+std::vector <std::map<std::string, std::vector<std::string> > > ModuleDependencies::getBuildRequires()
+{
+    auto cBuildRequires = modulemd_dependencies_peek_buildrequires(dependencies);
+    return getRequirements(cBuildRequires);
+
+}
+
+std::vector <std::map<std::string, std::vector<std::string> > > ModuleDependencies::getRequires()
+{
+    auto cRequires = modulemd_dependencies_peek_requires(dependencies);
+    return getRequirements(cRequires);
+}
+
+std::map<std::string, std::vector<std::string>> ModuleDependencies::wrapModuleDependencies(const void *moduleName, const void *streams) const
+{
+    std::map<std::string, std::vector<std::string> > moduleRequirements;
+
+    auto name = static_cast<const char *>(moduleName);
+    auto streamSet = modulemd_simpleset_dup((ModulemdSimpleSet *) streams);
+    while (streamSet != nullptr) {
```

Comment:
Are you waiting until the pointer reaches the nullptr value? Good luck.
And memory is finite in my notebook. Please free unused memory.
Look at https://github.com/jrohel/libdnf/commit/76b4bfe3df0439556d491c4f5cce72239bcf3e69 for inspiration.